activesupport/lib/active_support/callbacks.rb

```diff
@@ -373,7 +373,7 @@ def initialize(name, filter, kind, options, chain_config)
       def filter; @key; end
       def raw_filter; @filter; end
 
-      def merge(chain, new_options)
+      def merge_conditional_options(chain, new_options)
```

Comment:
As we are here, and `merge_conditional_options` `new_options` parameter can only have a `if` or `unless` option , I wonder if we should change this to keyargs: something like
```
def merge_conditional_options(chain, if: [], unless: [])
```

Not strong about tho, just a thought... @rafaelfranca WDYT?